python/lsst/cell_coadds/_fits.py

```diff
@@ -0,0 +1,407 @@
+                    band=hdu_list[1].header["BAND"],
+                ),
+            )
+
+            grid_cell_size = Extent2I(hdu_list[1].header["GRCELL1"], hdu_list[1].header["GRCELL2"])
+            grid_shape = Extent2I(hdu_list[1].header["GRSHAPE1"], hdu_list[1].header["GRSHAPE2"])
+            grid_min = Point2I(hdu_list[1].header["GRMIN1"], hdu_list[1].header["GRMIN2"])
+            grid = UniformGrid(cell_size=grid_cell_size, shape=grid_shape, min=grid_min)
+
+            outer_cell_size = Extent2I(hdu_list[1].header["OCELL1"], hdu_list[1].header["OCELL2"])
+
+            psf_image_size = Extent2I(hdu_list[1].header["PSFSIZE1"], hdu_list[1].header["PSFSIZE2"])
+
+            inner_bbox = Box2I(
+                Point2I(hdu_list[1].header["INBBOX11"], hdu_list[1].header["INBBOX12"]),
+                Point2I(hdu_list[1].header["INBBOX21"], hdu_list[1].header["INBBOX22"]),
+            )
+
+            header = hdu_list[1].header
+            coadd = MultipleCellCoadd(
+                (self._readSingleCellCoadd(data=row, header=header, common=common) for row in data),
+                grid=grid,
+                outer_cell_size=outer_cell_size,
+                psf_image_size=psf_image_size,
+                inner_bbox=inner_bbox,
+                common=common,
+            )
+
+        return coadd
+
+    def readAsExplodedCoadd(self) -> ExplodedCoadd:
+        """Read the FITS file as an ExplodedCoadd object."""
+        coadd = self.readAsMultipleCellCoadd()
+        with fits.open(self.filename) as hdu_list:
+            header = hdu_list[1].header
+        pad_psfs_with = header["PADPSF"]
+        return coadd.explode(pad_psfs_with=pad_psfs_with)
+
+    def readAsStitchedCoadd(self) -> StitchedCoadd:
+        """Read the FITS file as a StitchedCoadd object."""
+        coadd = self.readAsMultipleCellCoadd()
+        with fits.open(self.filename) as hdu_list:
+            header = hdu_list[1].header
+        minY = header["STBBOX12"]
+        minX = header["STBBOX11"]
+        maxX = header["STBBOX21"]
+        maxY = header["STBBOX22"]
+        bbox = Box2I(Point2I(minX, minY), Point2I(maxX, maxY))
+        return coadd.stitch(bbox=bbox)
+
+    @staticmethod
+    def _readSingleCellCoadd(
+        data: Mapping[str, Any], header: Mapping[str, Any], common: CommonComponents
+    ) -> SingleCellCoadd:
+        """Read a coadd from a FITS file.
+
+        Parameters
+        ----------
+        data : `Mapping`
+            The data from the FITS file.
+        header : `dict`
+            The header of the FITS file.
+        common : `CommonComponents`
+            The common components of the coadd.
+
+        Returns
+        -------
+        coadd : `SingleCellCoadd`
+            The coadd read from the file.
+        """
+        buffer_x, buffer_y = (header["OCELL1"] - header["GRCELL1"]) // 2, (
+            header["OCELL2"] - header["GRCELL2"]
+        ) // 2
```

Comment:
Partly thanks to black, this would read better as two separate statements.

But I think it'd be even better to pass the `UniformGrid` object (and any other necessary `Point2I` or `Extent2I` objects) in instead of passing the header.  That will keep the mapping from FITS keyword strings to `UniformGrid` concepts limited to just two places (one for read, one for write).

Reply:
Makes sense. I've removed passing `header` here and pass in only the relevant objects.